code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -193,6 +193,12 @@
 							L[tmpname] = R
 					var/desc = input("Please select a telepad.", "RCS") in L
 					E.pad = L[desc]
+					if(!Adjacent(user))
+						to_chat(user, "<span class='notice'>Unable to teleport, too far from crate.</span>")
+						return
+					if(!(E.rcell && (E.rcell.charge >= E.chargecost)))

```

Comment:
Do we want to tell the user they don't have enough charge before making them wait through a do_after, or do we want them to wait through a do_after only to find out they've insufficient charge?